src/algorithms/GenericFunctions.jl

```diff
@@ -118,7 +118,7 @@ function powermod(a::T, n::Integer, m::T) where T <: RingElem
       return internal_powermod(a, n, m)
    elseif n == 1
```

Comment:
You might as well also do 
```suggestion
   elseif isone(n)
```